tabpfn_time_series/features/feature_pipeline.py

```diff
@@ -0,0 +1,470 @@
+        self.train_df = None
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        Detects seasonal periods from the target time series.
+
+        This method analyzes the provided time series (`y` or `X['target']`)
+        to find the most significant seasonal periods using FFT and stores
+        them for the `transform` step.
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            The input data. If `y` is None, `X` must contain a 'target' column.
+        y : Optional[pd.Series], optional
+            The target time series, by default None.
+
+        Returns
+        -------
+        self
+            The fitted estimator.
+        """
+        # save the train_df
+        self.train_df = X
+
+        if y is None:
+            if "target" not in X.columns:
+                raise ValueError(
+                    "Target column 'target' not found in X. "
+                    "Please provide the target series via the y argument."
+                )
+            target_values = X["target"]
+        else:
+            target_values = y
+
+        detected_periods_and_magnitudes = self._find_seasonal_periods(
+            target_values=target_values,
+            max_top_k=self.max_top_k,
+            do_detrend=self.do_detrend,
+            detrend_type=self.detrend_type,
+            use_peaks_only=self.use_peaks_only,
+            apply_hann_window=self.apply_hann_window,
+            zero_padding_factor=self.zero_padding_factor,
+            round_to_closest_integer=self.round_to_closest_integer,
+            validate_with_acf=self.validate_with_acf,
+            sampling_interval=self.sampling_interval,
+            magnitude_threshold=self.magnitude_threshold,
+            relative_threshold=self.relative_threshold,
+            exclude_zero=self.exclude_zero,
+        )
+
+        self.periods_ = [period for period, _ in detected_periods_and_magnitudes]
+
+        return self
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Adds seasonal features (sine and cosine) to the DataFrame.
+
+        This method uses the periods detected during the `fit` phase to
+        generate and append seasonal features to the input DataFrame.
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            The input data to transform.
+
+        Returns
+        -------
+        pd.DataFrame
+            The DataFrame with the added seasonal features.
+        """
+        if not hasattr(self, "periods_"):
+            raise RuntimeError(
+                "The transformer has not been fitted yet. "
+                "Please call 'fit' before 'transform'."
+            )
+
+        X_transformed = X.copy()
+        if len(X_transformed["target"]) == len(self.train_df["target"]):
+            time_idx = np.arange(len(X_transformed))
+        else:
+            time_idx = np.arange(len(X_transformed))
+            time_idx += len(self.train_df)
```

Comment:
Fragile mechanism to differentiate `x_train` and `x_test`. See the comment above.